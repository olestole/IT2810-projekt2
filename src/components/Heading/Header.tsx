import React, { useContext } from 'react';
import moon from 'assets/header/moon.svg';
import sun from 'assets/header/sun.svg';
import AppContext from 'utils/AppContext';
import Logo from 'assets/header/appLogo.svg';

import './header.css';

const Heading = () => {
  const { appState, appDispatch } = useContext(AppContext);

  const handleHomebuttonClick = () => {
    appDispatch({ type: 'setCurrentUser', payload: null });
  };

  const toggleDarkmode = () => {
    appDispatch({ type: 'darkmode' });
  };

  return (
    <div>
      <header className="header">
        <div className="logo">
          <img src={Logo} onClick={handleHomebuttonClick}></img>
        </div>
        <div className="darkmode">
          <img src={appState.darkmode ? sun : moon} onClick={toggleDarkmode}></img>
        </div>
      </header>
    </div>
  );
};

export default Heading;
