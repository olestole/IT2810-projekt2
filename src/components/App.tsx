import React, { useContext } from 'react';
import AppContext from '../utils/AppContext';
import Heading from './Heading/Header';

import ProfileView from '../pages/ProfileView';
import GalleryView from '../pages/GalleryView';

import './main.css';

const App = () => {
  const { appState, appDispatch } = useContext(AppContext);

  return (
    <div className="App">
      <Heading />
      {appState.currentUser !== null ? <ProfileView /> : <GalleryView />}
    </div>
  );
};

export default App;
