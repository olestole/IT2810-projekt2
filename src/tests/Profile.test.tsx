import React from 'react';
import { render, act, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import App from 'components/App';
import Header from '../components/Heading/Header';

describe('Profile test-suite', () => {
  test('profile renders', () => {
    //render(<Header />);
    const wrapper = <Header></Header>;
    expect(wrapper).toMatchSnapshot();
    //expect(screen.getByText('')).toBeInTheDocument();
  });
});
