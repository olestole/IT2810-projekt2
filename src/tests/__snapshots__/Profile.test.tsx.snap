// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test-suite profile renders 1`] = `<App />`;

exports[`Header test-suite profile renders 2`] = `<Heading />`;
